libs/api/src/main/java/de/cyface/api/Authorizer.java

```diff
@@ -91,45 +113,72 @@ public void handle(final RoutingContext context) {
+                        context.fail(UNAUTHORIZED, r.cause());
+                        return;
+                    }
+
+                    // Before async operations, pause request body parsing to not lose the body or protocol upgrades.
+                    if (pauseAndResume) {
+                        if (!parseEnded) {
+                            request.pause();
+                        }
+                    }
+                    // Load principal from authentication result, so it also contains the roles
+                    final var loadUsers = loadAccessibleUsers(r.result().principal());
                     loadUsers.onSuccess(accessibleUsers -> {
                         try {
-                            LOGGER.trace("Export request for {} authorized to access users {}", username, accessibleUsers);
+                            if (pauseAndResume && !parseEnded && !upgraded) {
+                                request.resume();
+                            }
+                            LOGGER.trace("Request for {} authorized to access users {}", username, accessibleUsers);
                             handleAuthorizedRequest(context, accessibleUsers, headers);
                         } catch (RuntimeException e) {
                             context.fail(e);
                         }
                     });
                     loadUsers.onFailure(e -> {
                         LOGGER.error("Loading accessible users failed for user {}", username, e);
-                        context.fail(500);
+                        context.fail(500, e);
                     });
-                } catch (final RuntimeException e) {
-                    context.fail(500, e);
+                } catch (RuntimeException e) {
+                    context.fail(e);
                 }
             });
-            authentication.onFailure(e -> {
-                LOGGER.error("Authorization failed for user {}!", username);
-                context.fail(401, e);
-            });
         } catch (final NumberFormatException e) {
             LOGGER.error("Data was not parsable!");
-            context.fail(422, e);
+            context.fail(ENTITY_UNPARSABLE, e);
         }
     }
 
     /**
      * This is the method that should be implemented by subclasses to carry out the business logic on an authorized
      * request.
-     *
+     * 
      * @param ctx Vert.x request context
-     * @param usernames A list of usernames for which the request is authorized to export data
+     * @param users A list of usernames for which the request is authorized to export data
      * @param header the header of the request which may contain parameters required to process the request.
      */
-    protected abstract void handleAuthorizedRequest(final RoutingContext ctx, final List<String> usernames,
+    protected abstract void handleAuthorizedRequest(final RoutingContext ctx, final List<User> users,
```

Comment:
As we now also need the `userId`, I created a new model `User` which contains both, userId  and userName.

We have to access the user database in order to get the userId as the mongo Auth principle only contains the username.